contracts/extensions/Auctions/DutchAuction.sol

```diff
@@ -0,0 +1,483 @@
+      collection,
+      item,
+      startingPrice,
+      endingPrice,
+      block.timestamp + _duration,
+      _memo
+    );
+  }
+
+  /**
+   * @notice Places a bid on an existing auction. Refunds previous bid if needed. The contract will only store the highest bid. The bid can be below current price in anticipation of the auction eventually reaching that price. The bid must be at or above the end price.
+   *
+   * @param collection ERC721 contract.
+   * @param item Token id to bid on.
+   */
+  function bid(
+    IERC721 collection,
+    uint256 item,
+    string calldata _memo
+  ) external payable override nonReentrant {
+    bytes32 auctionId = keccak256(abi.encodePacked(collection, item));
+    DutchAuctionData memory auctionDetails = auctions[auctionId];
+
+    if (auctionDetails.info == 0) {
+      revert INVALID_AUCTION();
+    }
+
+    uint256 expiration = uint256(uint64(auctionDetails.prices >> 192));
+
+    if (block.timestamp > deploymentOffset + expiration) {
+      revert AUCTION_ENDED();
+    }
+
+    if (auctionDetails.bid != 0) {
+      uint256 currentBidAmount = uint96(auctionDetails.bid >> 160);
+      if (currentBidAmount >= msg.value) {
+        revert INVALID_BID();
+      }
+
+      payable(address(uint160(auctionDetails.bid))).transfer(currentBidAmount);
+    } else {
+      uint256 endingPrice = uint256(uint96(auctionDetails.prices >> 96));
+      // TODO: consider allowing this as a means of caputuring market info
+      if (endingPrice > msg.value) {
+        revert INVALID_BID();
+      }
+    }
+
+    uint256 newBid = uint256(uint160(msg.sender));
+    newBid |= uint256(uint96(msg.value)) << 160;
+
+    auctions[auctionId].bid = newBid;
+
+    // TODO: consider allowing a bid over the current minimum price to tranfer the token immediately without calling settle
+
+    emit PlaceBid(msg.sender, collection, item, msg.value, _memo);
+  }
+
+  /**
+   * @notice Settles the auction after expiration if no valid bids were received by sending the item back to the seller. If a valid bid matches the current price at the time of settle call, the item is sent to the bidder. Proceeds will be distributed separately by calling `distributeProceeds`.
+   *
+   * @param collection ERC721 contract.
+   * @param item Token id to settle.
+   */
+  function settle(
+    IERC721 collection,
+    uint256 item,
+    string calldata _memo
+  ) external override nonReentrant {
+    bytes32 auctionId = keccak256(abi.encodePacked(collection, item));
+    DutchAuctionData memory auctionDetails = auctions[auctionId];
+
+    if (auctionDetails.info == 0) {
+      revert INVALID_AUCTION();
+    }
+
+    uint256 lastBidAmount = uint256(uint96(auctionDetails.bid >> 160));
+    uint256 minSettlePrice = currentPrice(collection, item);
+
+    if (minSettlePrice < lastBidAmount) {
+      address buyer = address(uint160(auctionDetails.bid));
+
+      collection.transferFrom(address(this), buyer, item);
+
+      emit ConcludeAuction(
```

Comment:
Should the auctions not be deleted or closed somehow? Or is the emit enough to signal it has closed

Reply:
i wanted to separate auction settlement for the winner from the payment processing for the seller. for transaction fee reasons. `distributeProceeds` deals with the latter.